src/Microsoft.OpenApi.Readers/V2/OpenApiExternalDocsDeserializer.cs

```diff
@@ -26,7 +26,11 @@ internal static partial class OpenApiV2Deserializer
                 {
                     "url", (o, n) =>
                     {
-                        o.Url = new Uri(n.GetScalarValue());
+                        Uri uri;
+                        if (Uri.TryCreate(n.GetScalarValue(), UriKind.RelativeOrAbsolute, out uri))
+                        {
+                            o.Url = uri;
+                        }
```

Comment:
I'm not sure I'm a fan of this approach.  We are going to swallow the error and when reporting the problem all we can say is "you are missing a URL". This is going to be confusing to the person who believes they put a URL and it is invalid.  I think it is valid to have two distinct sources of errors.  One from parsing content, and one from validating the DOM.  We can't avoid it anyway because sometimes the YAML/JSON is going to be badly formatted.

Reply:
We should still produce the document though. If I throw an error here, everything breaks.

We need to have a way to still allow reading to go through and add the errors along the way.

---
In reply to: [165092404](https://github.com/Microsoft/OpenAPI.NET/pull/197#discussion_r165092404) [](ancestors = 165092404)

Reply:
Without that mechanism, what we can do is making the reporting message a bit more generic:

"The string passed in cannot be parsed into a URL."

This should be a very rare case. Almost any string can be a url. Even "abcdef" is allowed to be parsed in as a URL. Cases I suspect will be caught would be something like an empty string or a white space string.

---
In reply to: [167382549](https://github.com/Microsoft/OpenAPI.NET/pull/197#discussion_r167382549) [](ancestors = 167382549,165092404)